### DISCUSSION TITLE
Setting a target's required nullable property to a default value when the source doesn't have a property

### DISCUSSION THREAD

**Discussion by sander1095** [riok/mapperly#1383] [Q&A]
Hello!

Apologies for the confusing title. The code will be a lot clearer.

My target class has a required, nullable property but the source doesn't have a property for this.

I can not figure out how to make mapperly work with this. In this case, I want to be able to configure mapperly to set it to the default value.

```csharp
public class Source
{
    public required string Text { get; set; }
}

public class Target
{
    public required string? Id { get; set; }
    public required string Text { get; set; }
}

[Mapper]
public static partial class Mapper
{
    // TODO: How can I tell Mapperly to initialize Target.Id with its default value (null)?
    // I do not have a value to map from!
    public static partial Target Map(this Source source);
}
```

Thanks!

**Comment by latonz** [comment]
Mapperly 4.0 introduces `MapValue` which is exactly what you need here. You can try it out with the latest version of Mapperly from the next channel `4.0.0-next.1`. See the [docs](https://next.mapperly.riok.app/docs/configuration/constant-generated-values/#constant-values). In your case it should work by applying `[MapValue(nameof(Target.Id), null)]` to the `Map` method.

Note that this is a prerelease and it may break its API with any release.
